Draft-Accepted/RFC0066-PowerShell-User-Content-Location.md

```diff
@@ -0,0 +1,133 @@
+so that I can better manage my environment and better secure and support my users,
+to reduce unnecessary support tickets to my IT Team.
+```
+
+- PowerShell currently places profile, modules, and configuration files in the user's Documents
+  folder, which is against established conventions for shell configurations and tools.
+- PowerShell content files in OneDrive can lead to unwanted syncing of module files, leading to
+  various issues.
+- There is strong community demand for changing this behavior as the current setup is problematic
+  for many users.
+- Changing the default location would align PowerShell with other developer tools and improve
+  usability.
+
+## Specification
+
+- This will be an experimental feature.
+- The content folder location change will only apply to PowerShell on Windows.
+- Configurability of the content folder will apply to all platforms.
+- A configuration file in the PowerShell user content folder will determine the location of the user
+  scoped **PSModulePath**.
+  - By default, the PowerShell user content folder will be located in the
+    `$env:LOCALAPPDATA\PowerShell`.
+  - The new location becomes the location used as the `CurrentUser` scope for PSResourceGet.
+- The proposed directory structure:
+
+  ```
+  C:\Users\UserName\AppData\Local\PowerShell\
+  ├── powershell.config.json   (Not Configurable)
+  ├── Scripts                  (Configurable)
+  ├── Modules                  (Configurable)
+  ├── Help                     (Configurable)
+  └── <*profile>.ps1           (Configurable)
+  ```
+
+- The following setting is added to the `powershell.config.json` file:
+
+  **UserPSContentPath** specifies the full path of the content folder. The default value is
+  `$env:LOCALAPPDATA\PowerShell`. The user can change this value to a different path.
+
+  ```json
+  {
+      "UserPSContentPath" : "$env:LOCALAPPDATA\\PowerShell",
+  }
+  ```
+
+## User Experience
+
+- On startup PowerShell will create a directory in AppData and a configuration file.
+- The user scoped **PSModulePath** will point to `Modules` folder under the location specified by
+  **UserPSContentPath**.
+- Users can configure a custom location for PowerShell user content by changing the value of
+  **UserPSContentPath**.
+- Users will initially need to manually move/copy their existing PowerShell user content 
+  from the Documents folder to the new location after enabling the feature.
+- We intend to add a new cmdlet to aid this process in a future release shortly 
+  after this work has been completed as part of the work into hopefully the 7.6.0 GA
+  release or another 7.6.x release. Details on this to follow.
+
+## Other considerations
+
+- The following functionalities will be affected:
+  - SecretManagement
+    - SecretManagement extension vaults are registered for the current user context in:
+      `$env:LOCALAPPDATA\Microsoft\PowerShell\secretmanagement\secretvaultregistry\`
```

Comment:
This is why we need to keep it in this `$env:LOCALAPPDATA\Microsoft\PowerShell\` path as to reduce configuration location sprawl & simplify documentation